libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -146,9 +148,13 @@ export class GRPCService implements Service {
   ): Promise<void> {
     return new Promise((resolve, reject) => {
       this.logger?.debug(`${FlagdProvider.name}: connecting stream, attempt ${this._streamConnectAttempt}...`);
-      const stream = this._client.eventStream({});
-      stream.on('error', () => {
-        this.handleError(reject, connectCallback, changedCallback, disconnectCallback);
+      const stream = this._client.eventStream({}, {});
+      stream.on('error', (err: ServiceError | undefined) => {
+        if (err?.code === status.CANCELLED) {
+          this.logger?.debug(`${FlagdProvider.name}: stream cancelled, will not be re-established`);
```

Comment:
Don't start a reconnect in the cancelled case, which puts us in a loop and hangs things because `close()` waits for the stream to be done.